data/store/mongo/mongo_devicetokens.go

```diff
@@ -0,0 +1,59 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/tidepool-org/platform/devicetokens"
+	structuredmongo "github.com/tidepool-org/platform/store/structured/mongo"
+)
+
+// deviceTokensRepo implements devicetokens.Repository, writing data to a
+// MongoDB collection.
+type deviceTokensRepo structuredmongo.Repository
+
+// Upsert will create or update the given Config.
+func (r *deviceTokensRepo) Upsert(ctx context.Context, doc *devicetokens.Document) error {
+	// The presence of UserID and TokenID should be enforced with a mongodb
+	// index, but better safe than sorry.
+	if doc.UserID == "" {
+		return fmt.Errorf("UserID may not be empty")
```

Comment:
Sure, didn't realize that was a thing.

Reply:
It is a thing because it existed before much of the current Golang error package incarnation. It adds a lot of useful functionality to an error that didn't exist in Golang when `platform` was originally created.